qiskit_research/utils/convenience.py

```diff
@@ -45,7 +47,12 @@ def add_dynamical_decoupling(
     pass_manager = PassManager(
         list(dynamical_decoupling_passes(backend, dd_str, scheduler))
     )
-    circuits_dd = pass_manager.run(circuits)
+    if isinstance(circuits, QuantumCircuit):
+        circuits_dd = pass_manager.run(circuits)
+    elif isinstance(circuits[0], QuantumCircuit):
+        circuits_dd = pass_manager.run(circuits)
+    else:
+        circuits_dd = [pass_manager.run(circs) for circs in circuits]
```

Comment:
I would prefer not to allow a list of lists because it deviates from Qiskit itself and introduces extra logic here. Could I ask that you simply write this list comprehension in your own code?

Reply:
Ok.